Assets/MixedRealityToolkit.Providers/Experimental/WindowsMixedReality/WindowsMixedRealitySpatialAwarenessSceneUnderstandingObserver.cs

```diff
@@ -0,0 +1,1245 @@
+            {
+                var saso = ConvertSceneObject(filteredSceneObjects[i]);
+                convertedObjects.Add(saso);
+            }
+
+            await Task.Yield();
+
+            return convertedObjects;
+        }
+
+        private void UpdateTimerEventHandler(object sender, ElapsedEventArgs args)
+        {
+            if (AutoUpdate)
+            {
+                observerState = ObserverState.GetScene;
+            }
+            return;
+        }
+
+        private void ReadProfile()
+        {
+            //Debug.Log("ReadProfile");
+
+            if (ConfigurationProfile == null)
+            {
+                return;
+            }
+
+            MixedRealitySpatialAwarenessSceneUnderstandingObserverProfile profile = ConfigurationProfile as MixedRealitySpatialAwarenessSceneUnderstandingObserverProfile;
+            if (profile == null)
+            {
+                Debug.LogError("Windows Mixed Reality Scene Understanding Observer's configuration profile must be a MixedRealitySceneUnderstandingObserverProfile.");
+                return;
+            }
+
+            StartupBehavior = profile.StartupBehavior;
+            AutoUpdate = profile.AutoUpdate;
+            DefaultMaterial = profile.DefaultMaterial;
+            DefaultWorldMeshMaterial = profile.DefaultWorldMeshMaterial;
+            SurfaceTypes = profile.SurfaceTypes;
+            RequestMeshData = profile.RequestMeshData;
+            RequestPlaneData = profile.RequestPlaneData;
+            CreateGameObjects = profile.CreateGameObjects;
+            UsePersistentObjects = profile.UsePersistentObjects;
+            UpdateInterval = profile.UpdateInterval;
+            FirstUpdateDelay = profile.FirstUpdateDelay;
+            ShouldLoadFromFile = profile.ShouldLoadFromFile;
+            SerializedScene = profile.SerializedScene;
+            InferRegions = profile.InferRegions;
+            WorldMeshLevelOfDetail = profile.WorldMeshLevelOfDetail;
+            InstantiationBatchRate = profile.InstantiationBatchRate;
+            ObservationExtents = profile.ObservationExtents;
+            QueryRadius = profile.QueryRadius;
+            RequestOcclusionMask = profile.RequestOcclusionMask;
+            OcclusionMaskResolution = profile.OcclusionMaskResolution;
+            OrientScene = profile.OrientScene;
+        }
+
+        private enum ObserverState
+        {
+            Idle = 0,
+            WaitForAccess,
+            GetScene,
+            GetSceneTransform,
+            Working
+        }
+        private ObserverState observerState = ObserverState.WaitForAccess;
+        private CancellationToken killToken;
+        private bool shouldAutoStart;
+        private bool isRemoting;
+        private Guid sceneOriginId;
+
+        private System.Numerics.Matrix4x4 sceneToWorldXformSystem;
+
+        private List<SceneObject> filteredSelectedSurfaceTypesResult = new List<SceneObject>(128);
+        private Texture defaultTexture;
```

Comment:
nit: can you put all of the private variables together near top of class below public properties to match convention of code base?